infra/gcp/main.tf

```
@@ -59,3 +59,12 @@ resource "generic_local_cmd" "kubeconfig" {
     EOT
   }
 }
+
+# Synchronises local DAG files with those in the Cloud Composer environment.
+resource "google_storage_bucket_object" "python_files" {
+  for_each = { for dag_file in local.dag_files : dag_file => dag_file }
```

You are giving a map to the `for_each` meta-argument, but since for each couple of `key:value`, `key `and `value` share the same value, you could rather give a set to `for_each`. See [for_each doc](https://developer.hashicorp.com/terraform/language/meta-arguments/for_each#basic-syntax)
```suggestion
  for_each = toset([for dag_file in local.dag_files : dag_file])
```

---

```suggestion
  for_each = toset(local.dag_files)
```